modules/component/src/core/cd-aware/cd-aware_creator.ts

```diff
@@ -1,36 +1,30 @@
-import { ChangeDetectorRef, NgZone } from '@angular/core';
 import {
+  BehaviorSubject,
+  combineLatest,
+  EMPTY,
   NextObserver,
   Observable,
   PartialObserver,
   Subject,
   Subscribable,
   Subscription,
 } from 'rxjs';
-import { distinctUntilChanged, map, switchAll, tap } from 'rxjs/operators';
-import { toObservableValue } from '../projections';
-import { getChangeDetectionHandler } from './get-change-detection-handling';
-
-export interface CoalescingConfig {
-  optimized: boolean;
-}
+import {
+  catchError,
+  distinctUntilChanged,
+  switchMap,
+  tap,
+} from 'rxjs/operators';
+import {
+  CdStrategy,
+  DEFAULT_STRATEGY_NAME,
+  StrategySelection,
+} from './strategy';
+import { nameToStrategy } from './nameToStrategy';
 
 export interface CdAware<U> extends Subscribable<U> {
-  next: (value: Observable<U> | Promise<U> | null | undefined) => void;
-}
-
-export interface WorkConfig {
-  context: any;
-  ngZone: NgZone;
-  cdRef: ChangeDetectorRef;
-}
-
-export function setUpWork(cfg: WorkConfig): () => void {
-  const render: (component?: any) => void = getChangeDetectionHandler(
-    cfg.ngZone,
-    cfg.cdRef
-  );
-  return () => render(cfg.context);
+  nextPotentialObservable: (value: any) => void;
```

Comment:
Had it in some version. The idea is, I can't know what value ends up in the template and I convert ObservableInputs and null/undefined. The rest throws.

I can adapt to it. What do you say?

Reply:
Of course no one knows what values ends up in the template, but with generics you say _"let's say the passed in value here would be of type `T` then here it would be `T` as well and over here it would be `Observable<T>`"_ and so on.

Reply:
Wil fix it

Reply:
Fixed